arch/arm/boot/dts/bcm270x.dtsi

```diff
@@ -164,6 +164,19 @@
 &gpio {
 	interrupts = <2 17>, <2 18>;
 
+	dpi_18bit_cpadhi_gpio0: dpi_18bit_cpadhi_gpio0 {
```

Comment:
A necessary changes to the base DTB must be done using your overlay - delete this.

Reply:
Adding them to 'bcm270x.dtsi' has been discussed here https://www.raspberrypi.org/forums/viewtopic.php?f=44&p=1879211&sid=876ab8a01243f66cef80cf878edc5b06#p1871105 and here https://www.raspberrypi.org/forums/viewtopic.php?f=44&t=310462#p1856903. That's why I ended up adding a new configuration to 'bcm2702x.dtsi.

Please let me know how to proceed with this.

Reply:
> not having it duplicated in any overlay that happens to need to use this mode.

I can't see this being used often, but I'll accept it anyway. However, the duplicate must be removed from the overlay.

Reply:
@pelwell: I assume you're speaking of 'dpi18cpadhi-overlay.dts'. right?  For that one I've followed the same approach as with 'dpi18-overlay.dts' and 'dpi24-overlay.dts'. Adding ''dpi18cpadhi-overlay.dtbo' to config.txt is - at least from my current understanding - mandatory for using the display with FKMS/legacy, having the timing parameters defined via 'dpi_output_format' and 'dpi_timing'. Confirmed working btw.

In case of KMS a different overlay is used; here I followed exactly the same as with the adafruit 7in Kippah.

Reply:
Yes - that overlay: https://github.com/raspberrypi/linux/pull/4400/files#diff-a5e6aa3ed6ce23e2a89888037dc6df738f9f420414e79cb592cf7fb6df0b6c8dR30

Adding the pinctrl definitions in both the base dts (via bcm2708.dtsi) and in the overlay is not necessary. I was initially suggesting that you only put them in the overlay, but we've now switched to only putting them in the base dts and only referring to the label (`&dpi_18bit_cpadhi_gpio0`) from the overlay.